frontend/awx/common/MultipleChoiceField.tsx

```diff
@@ -46,25 +48,40 @@ export function MultipleChoiceField(props: IProps) {
 
   const docsURL = `${getDocsBaseUrl(config)}/html/userguide/job_templates.html#surveys`;
 
+  const emptyChoice = t('Choice option cannot be empty.');
+
   const defaultOptLabel = t('Default option');
 
-  const { control, setValue } = useFormContext();
+  const { control, setValue, setError, clearErrors } = useFormContext();
 
   const { fields, append, remove, update, replace } = useFieldArray({
     control,
-    name: 'choices',
+    name: 'formattedChoices',
   });
 
   const choices = fields as ChoiceOption[];
-  const updatedChoices = useWatch({ name: 'choices' }) as ChoiceOption[];
+  const updatedChoices = (useWatch({ name: 'formattedChoices' }) as ChoiceOption[]) ?? [];
 
   const addChoice = (useWatch({ name: 'add-choice' }) as string) ?? '';
 
   const handleAdd = () => {
+    if (addChoice.length <= 0) {
+      setError('add-choice', { message: emptyChoice });
+      return;
+    }
+
     append({ name: addChoice, default: false });
     setValue('add-choice', '');
+    clearErrors('add-choice');
   };
 
+  useEffect(() => {
+    if (type === 'multiplechoice')
+      replace(updatedChoices.map((choice) => ({ ...choice, default: false })));
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [type]);
```

Comment:
When I go to edit a multiple choice (single select) question with a default value, the default value doesn't auto-select. I believe this chunk of code is flipping the default to false.

Reply:
Good catch, thank you, I'm a little surprised component test didn't catch this. I implemented this because if multiple default options are selected in multi select type and then user switches to single select, multiple default options persist selected, which is wrong.

Reply:
Updated with fix.